### ISSUE TITLE
deploy to Maven from p2 repo instead of SDK

### ISSUE THREAD

**Issue by mickaelistria** [eclipse-platform/eclipse.platform.releng.aggregator#739]
The SDK misses 2 source bundles: org.eclipse.e4.tools.emf.ui.sources & org.eclipse.e4.ui.progress.sources . It's not causing issue in the SDK, but the Maven deployment script looks up source bundles in the SDK, and thus miss them. So when deploying to Maven Central, the deployment is blocked by the org.eclipse.e4.tools.emf.ui & org.eclipse.e4.ui.progress Maven artifacts missing source attchment, which is requested on Central.

**Comment by mickaelistria** [comment]
A better solution. instead of hacking the SDK, would be to use he p2 repo as input when pushing to Maven instead of the SDK.
cc @SDawley

**Comment by merks** [comment]
This is no longer a problem.  The snapshot publishing build will fail when sources are missing and there are currently none missing, so this file is empty:

https://github.com/eclipse-platform/eclipse.platform.releng/blob/master/publish-to-maven-central/sourceBundles.txt

**Comment by mickaelistria** [comment]
I'm reopening as a followup of https://github.com/eclipse-platform/eclipse.platform.releng/pull/182

**Comment by merks** [comment]
The premise of this issue might well be a misunderstanding in the first place.  Nothing (no script) is "looking in the SDK" so I'm not sure what that statement means.  The problem lies in how the aggregator works for including "all" the contents of a repository, not in any scripts in the platform.  Specifically it considers "all contents" to be all features so when a source bundle is not mentioned in any feature, it's not part of the all contents.  It could be specifically listed in the requirements in the aggregator, but then one must list all the features explicitly which is also not very nice.   Given the direction of includeAllSources we are going in, I think it makes sense for the aggregator to automatically include the source bundle (or feature) for any bundle (or feature) that's otherwise included in the "all" contents.   I.e.,this issues should be addressed in the aggregator.  If you can live with broken Maven publishing for a view days I will find time to address the issue in the aggregator?  Okay?

Note that the builds aren't even really broken.  The artifact is still published, there's just no source jar for it:

https://repo.eclipse.org/content/repositories/eclipse-snapshots/org/eclipse/jdt/org.eclipse.jdt.jeview/1.4.200-SNAPSHOT/

**Comment by merks** [comment]
@mickaelistria As you can see, I've opened this to address the issue in the aggregator:

https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/739

I might even find time for this tomorrow, though it's not a huge rush, given the builds publish and that the missing source for the two new view bundles will only be a problem when publishing to Nexus...

**Comment by mickaelistria** [comment]
> The premise of this issue might well be a misunderstanding in the first place. Nothing (no script) is "looking in the SDK" so I'm not sure what that statement means.

Indeed, it is. I don't remember why and how I was driven to believe that the SDK zip was used as input for the conversion to Maven. But it was wrong anyway.
https://github.com/eclipse-cbi/p2repo-aggregator/issues/13 seems like a good solution. Let's close this one then.
